### PR TITLE
Non deterministic behavior fixed for plural forms and missing translations

### PR DESCRIPTION
**Pull request by borissuska**
Hi, we found a bug when plural forms are used. When you define plural form like this

``` javascript
  pluralForm: function(count) {
    if (count === 0) { return 'zero'; }
    if (count === 1) { return 'one'; }
    return 'other';
  }
```

and then use translation like this

``` javascript
  'pluralized.short-translation': {
    one: 'One Click',
    other: '{{count}} Clicks'
  },
```

It behaves non-deterministic when you're using count=0.

In case that translation never been used with any count value it returns Missing translation. In case that translation was previously used with count other than 0 it will return for count=0 last used translation. Which is sometimes "One Click", sometimes "0 Clicks".

I found a problem in Locale.findTranslation function and I added defaultPluralForm into config to determine which key should be used in case that pluralForm return value is missing in translation.

I also added test to test this non-deterministic behavior and test to test count=undefined feature (issue #332 and #383). I added sk and cs locale configuration too. I fixed phantomjs ES6 errors too.

### CODE REVIEW

**Comment by jamesarosen** [comment]
If the `pluralForm` function for a locale allows for `zero`, then the translations for that locale MUST define a `zero` form. I definitely want to fix the nondeterminism, though. ember-i18n should _always_ report a missing translation for these cases.

Regarding your `defaultPluralForm` solution:

If `n` is a number, it is guaranteed to fall into one of the locale's pluralization rules. If it's _not_ a number, pluralization doesn't really make sense. Your solution would emit something like `"undefined dogs"` when `count` is `undefined`. That seems like a bad experience for end-users, though I could see how it would help programmers find where they're assuming something is a number, but it's not yet loaded.

Thank you for finding this bug and describing it so clearly :)

**Comment by borissuska** [comment]
You're right, I removed `defaultPluralForm` because it disables notification about missing translation. Now I'm not sure what is correct behavior when you got previous pluralForm definition and following translation string:

```javascript
'pluralized.translation': '{{count}} Clicks'
```

Is this an invalid translation definition and for `{{t "pluralized.translation" count=100}}` it should return return "Missing translation" message or is it valid and for any count value it should return "... Clicks" string?

For now it returns "Missing translations: pluralized.translation" because of `other`, `one` and `zero` plural keys are not defined. I'm not sure if it is correct because of backward compatibility. Should I return "... Clicks" string in this case now? If yes, I'll add deprecation message too that developer have to define all locale plural forms for the translation key.

**Comment by jamesarosen** [comment]
```js
'pluralized.translation': '{{count}} Clicks'
```
is an ill-defined translation. It should be at minimum
```js
'pluralized.translation.other': '{{count}} Clicks'
```

**Comment by borissuska** [comment]
As I wrote before this was implemented as you expect.

Now, I improved Missing translation string. If you're looking for a pluralized.translation message with count parameter set, it will return "Missing translation: pluralized.translation.zero", "Missing translation: pluralized.translation.one" or "Missing translation: pluralized.translation.other" depending on pluralForm function for current locale and count value.

File: addon/services/i18n.js [hunk 1]
```diff
@@ -53,7 +53,7 @@ export default Parent.extend(Evented, {
     const count = get(data, 'count');
 
     const translation = locale.findTranslation(makeArray(key), count);
-    return typeOf(translation.result) !== 'undefined';
+    return typeOf(translation.result) !== 'undefined' && !translation.result._isMissing;
```
Comment (jamesarosen) [inline addon/services/i18n.js]: Can you rebase on top of master?

**Comment by jamesarosen** [comment]
jamesarosen/ember-i18n has been deprecated in favor of [ember-intl](https://github.com/ember-intl).

**Pull request closed without merging**